From 375f28d4d75b1665d3843820cb08c575395293d5 Mon Sep 17 00:00:00 2001
From: Yuya Hamamachi <yuya.hamamachi.sx@renesas.com>
Date: Thu, 30 Sep 2021 09:38:51 +0900
Subject: [PATCH 2/4] starterkit: linux: rom-writer: Improve userbility

Even if 'expect' becomes timeout, we cannot notice it
because process is not stopped.
This patch makes that process is stopped after showing error message
when timeout is occurred.

Signed-off-by: Yuya Hamamachi <yuya.hamamachi.sx@renesas.com>
---
 starterkit/linux/rom_writer | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/starterkit/linux/rom_writer b/starterkit/linux/rom_writer
index 4458399..29b9327 100755
--- a/starterkit/linux/rom_writer
+++ b/starterkit/linux/rom_writer
@@ -11,6 +11,13 @@ set ver   ""
 set mode  ""
 set tty   ""
 
+expect_before {
+    timeout {
+	puts "Timeout. Please retry.\r"
+	exit 1
+    }
+}
+
 proc argv_check {} {
     global argv
     global top
-- 
2.31.1

